catalog/core/catalog-core-encryption/src/main/java/org/codice/ddf/catalog/content/encryption/ContentEncryptorDecryptor.java

```diff
@@ -0,0 +1,275 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.encryption;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.io.ByteStreams;
+import ddf.catalog.content.encryption.EncryptorDecryptor;
+import ddf.security.SecurityConstants;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ContentEncryptorDecryptor implements EncryptorDecryptor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ContentEncryptorDecryptor.class);
+  private static final String CONTENT_ENCRYPTION_KEY_ALIAS_VALUE =
+      SecurityConstants.getContentEncryptionKeyAlias();
+  private static final String CONTENT_ENCRYPTION_KEY_PASSWORD_VALUE =
+      SecurityConstants.getContentEncryptionKeyPassword();
+  private static final String AES = "AES";
+  private static final String CIPHER_TYPE = "AES/GCM/NoPadding";
+  private static final String ENABLE_DEBUG_LOGGING_MSG =
+      " Enable debug logging for more information.";
+  private static final int KEY_LENGTH = 256;
```

Comment:
❓ Is this actually working with a 256bit key? it isn't supposed to unless you're doing unlimited strength encryption in the JVM. 128 should be the largest you can go.

Reply:
I may have updated my JDK to include the advanced security to allow for a 256 bit key.

Because A. I'm not sure how to verify I actually did that and B. I don't see decreasing the size of the key being a hard change, I'll plan on figuring out if this is an issue during the hero process.

Reply:
I went ahead and changed it to a 128 bit key.